CHANGELOG.md

```diff
@@ -9,3 +9,11 @@
 - feature: added hexadecimal conversion specifier
 - bugfix: `StringBuilder` initialization without parameter adds empty string to internal Values Array.
 - bugfix: When there is no placeholder in the template passed to String.Format it should return the original template
+
+#### 1.5.0
+- Using latest typescript version, please check if your environment still works with this version before upgrading!
+- feature: !IMPORTANT methods/properties starting with uppercase are marked deprecated now and will be removed in future versions! Check the deprecated infos
+- feature: added class `$String` for everybody who is facing issues when `String` is used.
```

Comment:
After looking over the changed index.ts I asked myself why there's the `String` class in the first place. All members are static anyways.

Maybe, if the exported `String` is an issue it would be a nice idea to only export the methods of that module, or am I missing something that requires the `String` as a class?

Reply:
well, we actually dont really "need" that class for anything. How would you implement it?

Reply:
I'd just omit the class-syntax and export those functions directly.


For example by changing this

```ts
const EOL = '\r\n';

export class $String {
    private static readonly regexNumber = /{(\d+(:\w*)?)}/g;
    public static empty = '';

    public static isNullOrWhiteSpace(value: string | null | undefined): boolean {
        // ...
    }

    public static join(delimiter: string, ...args: (string | object | Array<any>)[]): string {
       // ...
    }
}
```

to this

```ts
const EOL = '\r\n';
const regexNumber = /{(\d+(:\w*)?)}/g;

export const empty = '';

export function isNullOrWhiteSpace(value: string | null | undefined): boolean {
    // ...
}

export function join(delimiter: string, ...args: (string | object | Array<any>)[]): string {
    // ...
}
```

This would, however, change the way the functions are imported (**so we should consider a major release**):

```diff
- import { $String } from 'typescript-string-operations'
- let joined = $String.join(" - ", "foo", "bar")

+ import { join } from 'typescript-string-operations' 
+ let joined = join(" - ", "foo", "bar")
```

Maybe we should move this conversation to an issue, discuss it there and merge this PR :grin:

Reply:
Done, for now its compatible, but i agree that we should remove this `String` class for the next major relase.